gui/src/3D/controls/touchButton3D.ts

```diff
@@ -250,30 +249,37 @@ export class TouchButton3D extends Button3D {
             newPushDepth = Math.max(newPushDepth, value);
         });
 
-        if (newPushDepth == ButtonState.Press) {
-            if (previousPushDepth == ButtonState.Hover) {
+        this._firePointerEvents(newPushDepth, previousPushDepth, pointOnButton);

```

Comment:
nit: While you're here, I would call these `newPushState`/`updatedPushState`. `previousPushState`, to me the depth name implies a continuous value for how deeply the button is being pushed